src/modules/schnorrsig/tests_exhaustive_impl.h

```diff
@@ -0,0 +1,211 @@
+            unsigned char sig64[64];
+            int actual_k = k;
+            if (skip_section(iter++)) continue;
+            if (parities[k - 1]) actual_k = EXHAUSTIVE_TEST_ORDER - k;
+            /* Generate random messages until all challenges have been tried. */
+            while (e_count_done < EXHAUSTIVE_TEST_ORDER) {
+                secp256k1_scalar e;
+                secp256k1_rand256(msg32);
+                secp256k1_schnorrsig_challenge(&e, xonly_pubkey_bytes[k - 1], msg32, xonly_pubkey_bytes[d - 1]);
+                /* Only do work if we hit a challenge we haven't tried before. */
+                if (!e_done[e]) {
+                    secp256k1_scalar expected_s = (actual_k + e * actual_d) % EXHAUSTIVE_TEST_ORDER;
+                    unsigned char expected_s_bytes[32];
+                    secp256k1_scalar_get_b32(expected_s_bytes, &expected_s);
+                    /* Invoke the real function to construct a signature. */
+                    CHECK(secp256k1_schnorrsig_sign(ctx, sig64, msg32, &keypairs[d - 1], secp256k1_hardened_nonce_function_smallint, &k));
+                    /* The first 32 bytes must match the xonly pubkey for the specified k. */
+                    CHECK(memcmp(sig64, xonly_pubkey_bytes[k - 1], 32) == 0);
+                    /* The last 32 bytes must match the expected s value. */
+                    CHECK(memcmp(sig64 + 32, expected_s_bytes, 32) == 0);
+                    /* Don't retry other messages that result in the same challenge. */
+                    e_done[e] = 1;
+                    ++e_count_done;
+                }
+            }
+        }
+    }
+}
+
+
+static void test_exhaustive_schnorrsig(const secp256k1_context *ctx, const secp256k1_ge* group) {
+    secp256k1_keypair keypair[EXHAUSTIVE_TEST_ORDER - 1];
+    secp256k1_xonly_pubkey xonly_pubkey[EXHAUSTIVE_TEST_ORDER - 1];
+    int parity[EXHAUSTIVE_TEST_ORDER - 1];
+    unsigned char xonly_pubkey_bytes[EXHAUSTIVE_TEST_ORDER - 1][32];
+    unsigned i;
+
+    /* Verify that all invalid_pubkey_bytes are actually invalid. */
+    for (i = 0; i < NUM_INVALID_KEYS; ++i) {
+        secp256k1_xonly_pubkey pk;
+        CHECK(!secp256k1_xonly_pubkey_parse(ctx, &pk, invalid_pubkey_bytes[i]));
+    }
+
+    /* Construct keypairs and xonly-pubkeys for the entire group. */
+    for (i = 1; i < EXHAUSTIVE_TEST_ORDER; ++i) {
+        secp256k1_scalar scalar_i;
+        unsigned char buf[32];
+        secp256k1_scalar_set_int(&scalar_i, i);
+        secp256k1_scalar_get_b32(buf, &scalar_i);
+        CHECK(secp256k1_keypair_create(ctx, &keypair[i - 1], buf));
+        CHECK(secp256k1_keypair_xonly_pub(ctx, &xonly_pubkey[i - 1], &parity[i - 1], &keypair[i - 1]));
+        CHECK(secp256k1_xonly_pubkey_serialize(ctx, xonly_pubkey_bytes[i - 1], &xonly_pubkey[i - 1]));
+    }
+
+    /* The keypairs/pubkey functions above have already been tested agains the group;
+       see src/modules/extrakeys/tests_exhaustive_impl.h. */
+    (void)group;
```

Comment:
why not just remove this argument?

Reply:
Yeah, done.

I wanted to keep the same "interface" for all exhaustive tests that are delegated to modules, but that's easy to add if we'd a lot more modules too.